spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc

```diff
@@ -869,6 +869,85 @@ The generated package is uploaded to Skipper's package repository and Data Flow
+ lifecycle with support of multi-versioned Stream App.
+This allows to upgrade or rollback the Stream Apps at runtime using the deployment properties.
+
+Register a Stream App with the versioned App Registry using the `app register` command. You must provide a unique name, application type, and a URI that can be resolved to the app artifact.
+For the type, specify "source", "processor", or "sink". The version is resolved from the URI. Here are a few examples:
+[source,bash]
+----
+dataflow:>app register --name mysource --type source --uri maven://com.example:mysource:0.0.1
+dataflow:>app register --name mysource --type source --uri maven://com.example:mysource:0.0.2
+dataflow:>app register --name mysource --type source --uri maven://com.example:mysource:0.0.3
+
+dataflow:>app list --id source:mysource
+╔══════════════════╤═════════╤════╤════╗
+║     source       │processor│sink│task║
+╠══════════════════╪═════════╪════╪════╣
+║> mysource-0.0.1 <│         │    │    ║
+║mysource-0.0.2    │         │    │    ║
+║mysource-0.0.3    │         │    │    ║
+╚══════════════════╧═════════╧════╧════╝
+----
+
+Depends on the schema the provided a URI format should conform to the following:
+
+* maven schema
+[source,bash]
+----
+maven://<groupId>:<artifactId>[:<extension>[:<classifier>]]:<version>
+----
+* http schema
+[source,bash]
+----
+http://<web-path>/<artifactName>-<version>.jar
+----
+* file schema
+[source,bash]
+----
+file:///<local-path>/<artifactName>-<version>.jar
+----
+
+You can mark the default app version to be used by when deploying Streams. By default the first registered app is marked as default.
+You can alter this using the `app default` command.
+
+[source,bash]
+----
+dataflow:>app default --id source:mysource --version 0.0.2
+dataflow:>app list --id source:mysource
+╔══════════════════╤═════════╤════╤════╗
+║     source       │processor│sink│task║
+╠══════════════════╪═════════╪════╪════╣
+║mysource-0.0.1    │         │    │    ║
+║> mysource-0.0.2 <│         │    │    ║
+║mysource-0.0.3    │         │    │    ║
+╚══════════════════╧═════════╧════╧════╝
+----
+
+Use the `app list --id <type:name>` command to list all version for a given Stream App.
+
+Use the `--version` option to unregister a particular app version.
+[source,bash]
+----
+dataflow:>app unregister --name mysource --type source --version 0.0.1
+dataflow:>app list --id source:mysource
+╔══════════════════╤═════════╤════╤════╗
+║     source       │processor│sink│task║
+╠══════════════════╪═════════╪════╪════╣
+║> mysource-0.0.2 <│         │    │    ║
+║mysource-0.0.3    │         │    │    ║
+╚══════════════════╧═════════╧════╧════╝
+----
+When the version is not specified the app registration version marked as default is deleted. Note: You have to manually set another version as default!
+
+Deployed or updated streams use the App versions marked as default unless a different version is provided with the set in the deployment properties.
```

Comment:
>  unless a different version is provided with the set in the deployment properties.

We could drop "with the set"

Reply:
Maybe also add an example of this experience? 

something like:

> stream create foo --definition "mysource | log" (default here is 0.0.2 based on the previous actions ^^)
> stream update foo --properties version.mysource=0.0.3

Reply:
+1